src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -204,48 +221,146 @@ void IODataBatchOperationListener.BatchOperationContentStreamDisposed()
+        /// <param name="headers">The headers for this request message.</param>
+        /// <param name="contentId">The contentId of this request message.</param>
+        /// <returns>The <see cref="ODataBatchOperationRequestMessage"/> instance.</returns>
+        protected ODataBatchOperationRequestMessage BuildOperationRequestMessage(
+            Func<Stream> streamCreatorFunc,
+            string method,
+            Uri requestUri,
+            ODataBatchOperationHeaders headers,
+            string contentId)
+        {
+            Uri uri = BuildOperationRequestUri(requestUri, this.inputContext.MessageReaderSettings.BaseUri);
+            this.contentIdToAddOnNextRead = contentId;
+
+            return new ODataBatchOperationRequestMessage(streamCreatorFunc, method, uri, headers, this,
+                contentId, this.payloadUriConverter, /*writing*/ false, this.container);
+        }
+
+        /// <summary>
+        /// Instantiate an <see cref="ODataBatchOperationResponseMessage"/> instance and set the status code.
+        /// </summary>
+        /// <param name="streamCreatorFunc">The function for stream creation.</param>
+        /// <param name="statusCode">The status code for the response.</param>
+        /// <param name="headers">The headers for this response message.</param>
+        /// <param name="contentId">The contentId of this request message.</param>
+        /// <returns>The <see cref="ODataBatchOperationResponseMessage"/> instance.</returns>
+        protected ODataBatchOperationResponseMessage BuildOperationResponseMessage(
+            Func<Stream> streamCreatorFunc,
+            int statusCode,
+            ODataBatchOperationHeaders headers,
+            string contentId)
+        {
+            ODataBatchOperationResponseMessage responseMessage = new ODataBatchOperationResponseMessage(
+                streamCreatorFunc, headers, this, contentId,
+                this.payloadUriConverter.BatchMessagePayloadUriConverter, /*writing*/ false, this.container);
+            responseMessage.StatusCode = statusCode;
+            return responseMessage;
+        }
+
+        /// <summary>
+        /// Instantiate an <see cref="Uri"/> object.
+        /// </summary>
+        /// <param name="requestUri">The uri to process.</param>
+        /// <param name="baseUri">The base Uri to use.</param>
+        /// <returns>An URI to be used in the request line of a batch request operation.
+        /// In the default scheme, the method either returns the specified <paramref name="requestUri"/> if it was absolute,
+        /// or it's combination with the <paramref name="baseUri"/> if it was relative.</returns>
+        /// <remarks>
+        /// This method will fail if no custom resolution is implemented and the specified <paramref name="requestUri"/> is
+        /// relative and there's no base URI available.
+        /// </remarks>
+        private Uri BuildOperationRequestUri(Uri requestUri, Uri baseUri)

```

Comment:
>BuildOperationRequestUri [](start = 20, length = 24)

Since this is a single line private method, should we just replace calls to BuildOperationRequestUri with calls to ODataBatchUtils.CreateOperationRequestUri(requestUri, baseUri, this.payloadUriConverter) and remove this method?

Reply:
OK, replaced with inline.

---
In reply to: [145847653](https://github.com/OData/odata.net/pull/960#discussion_r145847653) [](ancestors = 145847653)